### ISSUE TITLE
TensorFlow and TensorFlowLite model produces different results

### ISSUE THREAD

**Issue by pnovapps** [tensorflow/tensorflow#46142]
### System information

-   **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Simple test script included
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Google Colaboratory
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on a mobile device**:
N/A
-   **TensorFlow installed from (source or binary)**:
-   **TensorFlow version (use command below)**:
v2.4.0-0-g582c8d236cb 2.4.0
-   **Python version**:
-   **Bazel version (if compiling from source)**:
-   **GCC/Compiler version (if compiling from source)**:
-   **CUDA/cuDNN version**:
-   **GPU model and memory**:
-   **Exact command to reproduce**:
test script attached

### Describe the problem

I have a TensorFlow model (attached as a zip file) which is converted to TensorFlowLite using the attached script.
Both models are evaluated using a single test exemplar.
Expected: The two models should produce similar, if not identical, predictions.
Actual: The two models produce completely different predictions

(the tflite model produces the same 'wrong' predictions when running on a mobile device but the 
test can entirely reproduced in Colaboratory using the attached files).
[model.zip](https://github.com/tensorflow/tensorflow/files/5765363/model.zip)

[tflite_model_test.ipynb.txt](https://github.com/tensorflow/tensorflow/files/5765368/tflite_model_test.ipynb.txt)

### Source code / logs
Attached model.zip and tflite_model_test.ipynb

**Comment by abattery** [comment]
CCing @terryheo

**Comment by pnovapps** [comment]
I tracked down the cause of the issue to the use of a masking layer in the tensorflow model

  tf.keras.layers.Masking(mask_value=0., input_shape=(timesteps, features)),  
  
 removing this line from the sequential keras model and everything works as expected.

I didn't find anything saying that masking is not supported in tflite. If it is not supported the converter should display an error rather than creating an incorrect model.

**Comment by terryheo** [comment]
@pnovapps thanks for the information. We'll check.

@battery could you check why the converter didn't make any error on the masking layer?

**Comment by abattery** [comment]
I guess we need to figure which layer makes a wrong prediction, for example, kernel implementation or converter lowering pattern.

I will take a look at the attached notebook.

**Comment by abattery** [comment]
@pnovapps could you share the minimal reproducible steps for the problem with the sequential keras model? The notebook does not contain any keras code to build the model. It would be much helpful if you can give us in a form of the notebook and provide the example input and the expected output.

**Comment by pnovapps** [comment]
The previous script used the pre-trained attached model. I've attached a minimal script to train and evaluate the sequential keras model using random data. The model won't do anything useful but is sufficient to illustrate the differences in predictions. The script contains an example input, the script produces the actual and expected outputs.

On one run the tensorflow model predictions are
  0.07902689, 0.0826071 , 0.09351914, 0.11532243, 0.13965082, 0.07816847,
  0.07019586, 0.06621424, 0.08247119, 0.06793797, 0.12488594

and tensorflowlite predictions are
  0.08866501 0.12623808 0.05040342 0.07568947 0.13019517 0.0998802
  0.06874353 0.0756304  0.06352186 0.11585259 0.10518024

the differences are more pronounced with live data.
[tflite_model_test2.ipynb.txt](https://github.com/tensorflow/tensorflow/files/5801166/tflite_model_test2.ipynb.txt)

**Comment by abattery** [comment]
Thanks a lot! I can reproduce the problem.

**Comment by thaink** [comment]
I found the cause of this issue and fixed it at commit 13fa827d04cbe6de0e0f1f1cc190d06ce01aa074
